Merge/go ethereum 20200626

meowsbits opened the pull request.

tzdybal commented on params/types/genesisT/genesis.go:
```diff
@@ -520,6 +520,14 @@ func (g *Genesis) SetECIP1080Transition(n *uint64) error {
 }
 
+func (g Genesis) GetEIP1706Transition() *uint64 {
+	return g.Config.GetECIP1080Transition()
```
Wrong ECIP/EIP number ;)
```suggestion
	return g.Config.GetEIP1706Transition()
```

meowsbits replied:
Resolved with https://github.com/etclabscore/multi-geth/pull/126/commits/c51dc1efa03629f5156e3fd8be1983b107d788c9

duanhao0814 committed:
core: filter out txs with invalid signatures as soon as possible

Once we detect an invalid transaction during recovering signatures, we should
directly exclude this transaction to avoid validating the signatures hereafter.

This should optimize the validations times of transactions with invalid signatures
to only one time.

libby committed:
cmd/ethkey: support --passwordfile in generate command (#21183)

ucwong committed:
ethstats: use timer instead of time.sleep (#20924)

chfast committed:
core/vm: use uint256 in EVM implementation (#20787)

* core/vm: use fixed uint256 library instead of big

* core/vm: remove intpools

* core/vm: upgrade uint256, fixes uint256.NewFromBig

* core/vm: use uint256.Int by value in Stack

* core/vm: upgrade uint256 to v1.0.0

* core/vm: don't preallocate space for 1024 stack items (only 16)

Co-authored-by: Martin Holst Swende <martin@swende.se>

karalabe committed:
core/state: avoid escape analysis fault when accessing cached state

karalabe committed:
Merge pull request #21192 from karalabe/fix-escape-analysis

core/state: avoid escape analysis fault when accessing cached state

p4u committed:
accounts/keystore: fix typo in error message (#21200)

natsukagami committed:
accounts/abi: make GetType public again (#21157)

holiman committed:
eth/downloader: fix spuriously failing tests (#21149)

* eth/downloader tests: fix spurious failing test due to race between receipts/headers

* miner tests: fix travis failure on arm64

* eth/downloader: tests - store td in ancients too

rjl493456442 committed:
core, eth, internal: include read storage entries in structlog output (#21204)

* core, eth, internal: extend structLog tracer

* core/vm, internal: add storage view

* core, internal: add slots to storage directly

* core: remove useless

* core: address martin's comment

* core/vm: fix tests

howjmay committed:
build, internal/ethapi, crypto/bls12381: fix typos (#21210)

speicifc -> specific
assigened -> assigned
frobenious -> frobenius

holiman committed:
go.mod: update uint256 to 1.1.0 (#21206)

hackmod committed:
internal/web3ext: add missing params to debug.accountRange (#21208)

six-days committed:
eth/downloader: don't use defer for unlock before return (#21227)

Co-authored-by: linjing <linjingjing@baidu.com>

karalabe committed:
Merge pull request #21170 from duanhao0814/fix-dup-ecrecover

core: filter out txs with invalid signatures as soon as possible

zhangsoledad committed:
core/rawdb: swap tailId and itemOffset for deleted items in freezer (#21220)

* fix(freezer): tailId filenum offset were misplaced

* core/rawdb: assume first item in freezer always start from zero

rjl493456442 committed:
cmd, eth, internal, les: add txfee cap (#21212)

* cmd, eth, internal, les: add gasprice cap

* cmd/utils, eth: add default value for gasprice cap

* all: use txfee cap

* cmd, eth: add fix

* cmd, internal: address comments

ucwong committed:
go.mod: update snappy dependency (#21237)

ucwong committed:
whisper : use timer.Ticker instead of sleep (#21240)

* whisper : use timer.Ticker instead of sleep

* lint: Fix linter error

Co-authored-by: Guillaume Ballet <gballet@gmail.com>

AusIV committed:
core/rawdb: fix high memory usage in freezer (#21243)

The ancients variable in the freezer is a list of hashes, which
identifies all of the hashes to be frozen. The slice is being allocated
with a capacity of `limit`, which is the number of the last block
this batch will attempt to add to the freezer. That means we are
allocating memory for all of the blocks in the freezer, not just
the ones to be added.

If instead we allocate `limit - f.frozen`, we will only allocate
enough space for the blocks we're about to add to the freezer. On
mainnet this reduces usage by about 320 MB.

gballet committed:
common/fdlimit: build on DragonflyBSD (#21241)

* common/fdlimit: build on DragonflyBSD

* review feedback

ucwong committed:
whisper: fix time.sleep by time.ticker in whisper_test (#21251)

karalabe committed:
p2p: measure packet throughput too, not just bandwidth

karalabe committed:
Merge pull request #21256 from karalabe/p2p-packet-metrics

p2p: measure packet throughput too, not just bandwidth

fjl committed:
cmd/devp2p: add commands for node key management (#21202)

These commands mirror the key/URL generation functions of cmd/bootnode.

    $ devp2p key generate mynode.key
    $ devp2p key to-enode mynode.key -ip 203.0.113.21 -tcp 30304
    enode://78a7746089baf4b8615f54a5f0b67b22b1...

fjl committed:
crypto/secp256k1: enable 128-bit int code and endomorphism optimization (#21203)

* crypto/secp256k1: enable use of __int128

This speeds up scalar & field calculations a lot.

* crypto/secp256k1: enable endomorphism optimization

rjl493456442 committed:
eth: don't block if transaction broadcast loop fails (#21255)

* eth: don't block if transaction broadcast loop is returned

* eth: kick out peer if we failed to send message

* eth: address comment

MariusVanDerWijden committed:
trie: reduce allocs in insertPreimage (#21261)

MariusVanDerWijden committed:
core/vm: fix incorrect computation of BLS discount (#21253)

* core/vm: fix incorrect computation of discount

During testing on Yolov1 we found that the way geth calculates the discount
is not in line with the specification. Basically what we did is calculate
128 * Bls12381GXMulGas * discount / 1000 whenever we received more than 128 pairs
of values. Correct would be to calculate k * Bls12381... for k > 128.

* core/vm: better logic for discount calculation

* core/vm: better calculation logic, added worstcase benchmarks

* core/vm: better benchmarking logic

meowsbits committed:
cmd/geth: make import cmd exit with 1 if import errors occurred (#21244)

The import command should not return a 0 status
code if the import finishes prematurely becaues
of an import error.

Returning the error causes the program to exit with 1
if the err is non nil.

Signed-off-by: meows <b5c6@protonmail.com>

MariusVanDerWijden committed:
eth: returned revert reason in traceTx (#21195)

* eth: returned revert reason in traceTx

* eth: return result data

meowsbits committed:
Merge remote-tracking branch 'foundation/master'

 Conflicts:
	core/vm/contracts.go
	core/vm/eips.go
	core/vm/gas_table.go
	core/vm/instructions.go
	eth/gen_config.go
	go.sum

meowsbits committed:
vm: fix return values mismatch

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
vm: fix unused import

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
ethapi: fix undefined import path (params -> vars)

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.sum: updates go.sum

Signed-off-by: meows <b5c6@protonmail.com>
